package/lib/compileFunctions.js

```diff
@@ -145,12 +159,39 @@ const validateEventsProperty = (funcObject, functionName) => {
 
 const validateVpcConnectorProperty = (funcObject, functionName) => {
   if (funcObject.vpc && typeof funcObject.vpc === 'string') {
-    const vpcNamePattern = /projects\/[\s\S]*\/locations\/[\s\S]*\/connectors\/[\s\S]*/i;
-    if (!vpcNamePattern.test(funcObject.vpc)) {
+    // vpcConnector argument can be one of two possible formats as described here:
+    // https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#resource:-cloudfunction
+    if (funcObject.vpc.indexOf('/') > -1) {
+      const vpcNamePattern = /projects\/[\s\S]*\/locations\/[\s\S]*\/connectors\/[\s\S]*/i;
+      if (!vpcNamePattern.test(funcObject.vpc)) {
+        const errorMessage = [
+          `The function "${functionName}" has invalid vpc connection name`,
+          ' VPC Connector name should follow projects/{project_id}/locations/{region}/connectors/{connector_name}',
+          ' or just {connector_name} if within the same project.',
+          ' Please check the docs for more info at ',
+          ' https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#resource:-cloudfunction',
+        ].join('');
+        throw new Error(errorMessage);
+      }
+    }
+  }
+};
+
+const validateVpcConnectorEgressProperty = (funcObject, functionName) => {
+  if (funcObject.vpcEgress && typeof funcObject.vpcEgress === 'string') {
+    const egress = funcObject.vpcEgress.toUpperCase();
+    if (
```

Comment:
This check can be achieved by proper schema definition in for `vpcEgress` property